integration_tests/test_gravity_2.py

```diff
@@ -367,3 +368,62 @@ def check_gravity_native_tokens():
 
         # check that balance is still same
         assert cli.balance(eth_to_bech32(recipient), denom=denom) == max_int
+
+
+def test_gravity_non_cosmos_denom(gravity):
+    if not gravity.cronos.enable_auto_deployment:
+        cronos_cli = gravity.cronos.cosmos_cli()
+
+        # deploy test erc20 contract
+        erc20 = deploy_contract(
+            gravity.geth,
+            CONTRACTS["TestERC20A"],
+        )
+        print("send to cronos crc20")
+        recipient = HexBytes(ADDRS["community"])
+        balance = erc20.caller.balanceOf(ADDRS["validator"])
+        assert balance == 100000000000000000000000000
+        amount = 100
+        txreceipt = send_to_cosmos(
+            gravity.contract, erc20, gravity.geth, recipient, amount, KEYS["validator"]
+        )
+        assert txreceipt.status == 1, "should success"
+        assert erc20.caller.balanceOf(ADDRS["validator"]) == balance - amount
+
+        denom = f"gravity{erc20.address}"
+
+        def check_gravity_native_tokens():
+            "check the balance of gravity native token"
+            return cronos_cli.balance(eth_to_bech32(recipient), denom=denom) == amount
+
+        wait_for_fn("send-to-gravity-native", check_gravity_native_tokens)
+
+        # Deploy a bad cosmos erc20 token
```

Comment:
```suggestion
        # Deploy a bad cosmos erc20 token with single character
        # Cosmos denom must be 3 ~ 128 characters long and support letters, followed by either
	# a letter, a number or a separator ('/', ':', '.', '_' or '-').
```

Perhaps add comments why the denom is "bad". wdyt?